### ISSUE TITLE
Changing chain in the browser reloads the current page but new chain doesn't take effect

### ISSUE THREAD

**Issue by hboon** [AlphaWallet/alpha-wallet-android#2063]
1. Make sure Ethereum mainnet is selected in the browser
2. Visit https://omni.xdaichain.com/bridge
3. It says (paraphrased) "from ETH mainnet to xDai"
4. Change network to xDai
5. Page automatically reloads

Expected:

6. It says (paraphrased) "from xDai to ETH mainnet"

Observed:

6. It says (paraphrased) "from ETH mainnet to xDai"
7. Reload page
8. It says (paraphrased) "from xDai to ETH mainnet"

The fix might be #1914

**Comment by JamesSmartCell** [comment]
Yes this needs revisiting; the real fix was updating the provider, which now done, updating this should now work correctly.

**Comment by JamesSmartCell** [comment]
Works correctly now, possibly fixed since last release.

**Comment by hboon** [comment]
Still in v3.40.2 (171)
